### PR TITLE
Facebook Marketing documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Facebook Marketing extract source docs

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

**Comment by am6010** [comment]
Screenshot 2023-02-20 at 9 36 34 AM

**Review by am6010 — APPROVED** [review]

File: docs/sources/extract/facebook-marketing.mdx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+1. Click **Connect with Facebook Marketing** and grant RudderStack the necessary permissions to access your Facebook Marketing data.
+
+<div class="infoBlock">
+Your Facebook account and the related details will then automatically appear under <strong>Choose an account</strong>.
+</div>
+
+2. Then, configure the following settings:
+
+- **Account ID**: Enter your Facebook Ads account ID used to pull the data from the Marketing API.
```
Comment (ishachopra0401) [inline docs/sources/extract/facebook-marketing.mdx:37]:
```suggestion
- **Account ID**: Enter the Facebook Ads account ID you use to pull the data from the Marketing API.
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+- **Start Date**: Specify the date from which RudderStack replicates the data for all the incremental streams. RudderStack will **not** sync any historical data generated before this date.
```
Comment (ishachopra0401) [inline docs/sources/extract/facebook-marketing.mdx:38]:
```suggestion
- **Start Date**: Specify the date from which RudderStack should replicate the data for all the incremental streams. RudderStack will **not** sync any historical data generated before this date.
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+- **End Date**: RudderStack syncs all the data generated between **Start Date** and this date. If you do not specify this date, RudderStack will sync all the latest available data.
+- **Include Deleted**: Enable this setting to include the data from all the deleted campaigns, ads, and ad sets.
+- **Fetch Thumbnail Images**: Enable this setting to fetch and store the thumbnail data for each [Ad Creative](https://developers.facebook.com/docs/marketing-api/creative/).
+- **Custom Insights**: This setting lets you add a list containing the entries for ad insights entries. You can specify the following configuration settings by clicking the **Add Item** button:
```
Comment (ishachopra0401) [inline docs/sources/extract/facebook-marketing.mdx:42]:
```suggestion
- **Custom Insights**: This setting lets you add a list containing the ad insights entries. You can specify the following configuration settings by clicking the **Add Item** button:
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+### Selecting the data to import
+
+You can choose the Facebook Marketing data you want to ingest by selecting the required resources:
+
+<img src="../../assets/cloud-extract-sources/fb-marketing-resources.png" alt="Selecting the data to import" />
+
+The below table mentions the syncs supported by these resources from Facebook Marketing to your warehouse destination:  
+
+| Resource | Sync type | Facebook Marketing API endpoint |
```
Comment (ishachopra0401) [inline docs/sources/extract/facebook-marketing.mdx:83]: Till now, we have been mentioning only Incremental for Incremental+ Full refresh sync supporting resources. As we give the extra details [here](https://rudderstack-docs-git-facebook-marketing-source-rudder-stack.vercel.app/sources/extract/common-settings/#sync-modes:~:text=Any%20resource%20that%20supports%20the%20Incremental%20sync%20mode%20also%20supports%20Full%20Refresh%2C%20by%20default.%20You%20can%20contact%20the%20RudderStack%20team%20to%20sync%20your%20data%20using%20the%20Full%20Refresh%20mode%20in%20case%20your%20destination%27s%20data%20gets%20corrupted%20or%20you%20want%20to%20sync%20your%20data%20all%20over%20again.). This one can create confusion. LMK your thoughts.

File: docs/sources/extract/facebook-marketing.mdx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+| :---| :--- | :--- |
+| `activities` | Full Refresh, Incremental | [`ad_activity`](https://developers.facebook.com/docs/marketing-api/reference/ad-activity) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:85]:
```suggestion
| `activities` | Incremental | [`ad_activity`](https://developers.facebook.com/docs/marketing-api/reference/ad-activity) |
```

Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:85]:
```suggestion
| `activities` | Incremental | [`ad_activity`](https://developers.facebook.com/docs/marketing-api/reference/ad-activity) |
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+| `ad_account` |  Full Refresh |  |
+| `ad_creatives` | Full Refresh | [`ad-creative`](https://developers.facebook.com/docs/marketing-api/reference/ad-creative) |
+| `ad_sets` | Full Refresh, Incremental | [`ad-campaign`](https://developers.facebook.com/docs/marketing-api/reference/ad-campaign) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:88]:
```suggestion
| `ad_sets` | Incremental | [`ad-campaign`](https://developers.facebook.com/docs/marketing-api/reference/ad-campaign) |
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+| `ads` |  Full Refresh, Incremental | [`ad-group`](https://developers.facebook.com/docs/marketing-api/reference/adgroup) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:89]:
```suggestion
| `ads` |  Incremental | [`ad-group`](https://developers.facebook.com/docs/marketing-api/reference/adgroup) |
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+| `ads_insights` | Full Refresh, Incremental | [`insights`](https://developers.facebook.com/docs/marketing-api/reference/adgroup/insights/) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:90]:
```suggestion
| `ads_insights` | Incremental | [`insights`](https://developers.facebook.com/docs/marketing-api/reference/adgroup/insights/) |
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+| `campaigns` | Full Refresh, Incremental |  [`campaigns`](https://developers.facebook.com/docs/marketing-api/reference/ad-account/campaigns/) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:91]:
```suggestion
| `campaigns` | Incremental |  [`campaigns`](https://developers.facebook.com/docs/marketing-api/reference/ad-account/campaigns/) |
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+| `custom_conversions` | Full Refresh, Incremental | [`customconversions`](https://developers.facebook.com/docs/marketing-api/reference/custom-conversion) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:92]:
```suggestion
| `custom_conversions` | Incremental | [`customconversions`](https://developers.facebook.com/docs/marketing-api/reference/custom-conversion) |
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+| `images` | Full Refresh, Incremental | [`ad-image`](https://developers.facebook.com/docs/marketing-api/reference/ad-image) |
+| `videos` | Full Refresh, Incremental |  [`videos`](https://developers.facebook.com/docs/graph-api/reference/video/video_insights/) |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:94]:
```suggestion
| `videos` | Incremental |  [`videos`](https://developers.facebook.com/docs/graph-api/reference/video/video_insights/) |
```

File: docs/sources/extract/facebook-marketing.mdx (hunk 4 above)
Comment (ameypv-rudder) [reply docs/sources/extract/facebook-marketing.mdx:83]: Agreed - made the change

**Review by ishachopra0401 — APPROVED** [review]

File: docs/sources/extract/facebook-marketing.mdx [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+All the Cloud Extract sources support sending data only to a <Link to="/destinations/warehouse-destinations/">data warehouse destination</Link>.
+</div>
+
+<GhBadge
+  label={'Stability'}
+  message={'Beta'}
+  color={'blueviolet'}
+/>
+
+## Getting Started
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:19]:
```suggestion
## Getting started
```

File: docs/sources/extract/facebook-marketing.mdx [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+You can choose the Facebook Marketing data you want to ingest by selecting the required resources:
+
+<img src="../../assets/cloud-extract-sources/fb-marketing-resources.png" alt="Selecting the data to import" />
+
+The below table mentions the syncs supported by these resources from Facebook Marketing to your warehouse destination:  
+
+| Resource | Sync type | Facebook Marketing API endpoint |
+| :---| :--- | :--- |
+| `activities` | Incremental | [`ad_activity`](https://developers.facebook.com/docs/marketing-api/reference/ad-activity) |
+| `ad_account` |  Full Refresh |  |
```
Comment (ameypv-rudder) [inline docs/sources/extract/facebook-marketing.mdx:86]:
```suggestion
| `ad_account` |  Full Refresh | [Ad Account](https://developers.facebook.com/docs/marketing-api/business-asset-management/guides/ad-accounts) |
```

**Pull request merged**